src/main/java/com/gaggle/contact/controllers/ContactController.java

```diff
@@ -44,6 +46,20 @@
                 .orElseThrow(() -> new ContactNotFoundException(name));
     }
 
+    @GetMapping("/query")
+    public List<Contact> getByEmail(@RequestParam final String email) {
+        if (Objects.isNull(email) || !email.contains("@")) {
+            throw new RuntimeException("email is no good, try again silly");
```

Comment:
Instead of using an unchecked exception let's handle our code properly with better return logic or more specific errors, this messages isn't returning properly per requirement 1.